backend/fregepoc/indexers/models.py

```diff
@@ -73,3 +80,50 @@ def __iter__(self):
                 self.rate_limit_exceeded = True
                 break
             yield from repos_to_process
+
+
+class SourceforgeIndexer(SingletonModel, BaseIndexer):
+    current_page = models.PositiveIntegerField(
+        _("current page"),
+        default=1,
+        help_text=_("The last visited page."),
+    )
+
+    def __iter__(self):
+        singlePageProjectsExtractor = SinglePageProjectsExtractor()
+        singleProjectCodeUrlExtractor = SingleProjectCodeUrlExtractor()
+        singleProjectGitLinkExtractor = SingleProjectGitLinkExtractor()
+        singleProjectResponseExtractor = SingleProjectResponseExtractor()
+        singleProjectGitUrlExtractor = SingleProjectGitUrlExtractor()
+
+        for i in itertools.count(start=self.current_page):
+            projects = singlePageProjectsExtractor.extract(i)
+
+            self.current_page += 1
+            self.save(update_fields=["current_page"])
+
+            if not projects:
+                continue
```

Comment:
Is it a right way to handle empty pages? Won't it make the loop above infinite ?

Reply:
The current page was increment above this `if`, so IMO it will work ;)